src/java.desktop/share/classes/com/sun/imageio/plugins/jpeg/JPEGMetadata.java

```diff
@@ -252,6 +253,22 @@ public class JPEGMetadata extends IIOMetadata implements Cloneable {
                     newGuy.loadData(buffer);
                 }
                 break;
+            case JPEG.APP1:
+                buffer.loadBuf(8);
+                buf = buffer.buf;
+                ptr = buffer.bufPtr;
+                if ((buf[ptr+3] == 'E')
+                        && (buf[ptr+4] == 'x')
+                        && (buf[ptr+5] == 'i')
+                        && (buf[ptr+6] == 'f')
+                        && (buf[ptr+7] == 0)
+                        && (buf[ptr+8] == 0)) {
+                    newGuy = new ExifMarkerSegment(buffer);
```

Comment:
The `IllegalArgumentException` from the `ExifMarkerSegment` constructor should be handled somehow. Perhaps just fall back to setting `newGuy` to a plain `MarkerSegment`?

Reply:
This is updated (see this thread https://github.com/openjdk/jdk/pull/22898#discussion_r1955194737 )